3-0-spring-framework/3-3-0-enable-string-trimming/src/test/java/com/bobocode/TrimmingPostProcessorTest.java

```diff
@@ -0,0 +1,254 @@
+        Optional<Bean> beanAnnotation = Stream.of(methods)
+                .filter(m -> m.getReturnType().equals(TrimmedAnnotationBeanPostProcessor.class))
+                .map(m -> m.getAnnotation(Bean.class)).findFirst();
+        assertTrue(beanAnnotation.isPresent());
+    }
+
+    @Test
+    @Order(9)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor implements BeanPostProcessor interface")
+    void trimmedAnnotationBeanPostProcessor_implementsBeanPostProcessor() {
+        boolean isBeanPostProcessorPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getInterfaces())
+                .anyMatch(i -> i.isAssignableFrom(
+                        BeanPostProcessor.class));
+        assertTrue(isBeanPostProcessorPresent);
+    }
+
+    @Test
+    @Order(10)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor overrides postProcessAfterInitialization() method")
+    void trimmedAnnotationBeanPostProcessor_overridesPostProcessAfterInitMethod() {
+        List<String> methodNames = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getDeclaredMethods())
+                .map(Method::getName)
+                .toList();
+        assertThat(methodNames, contains("postProcessAfterInitialization"));
+    }
+
+    @Test
+    @Order(11)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor does not override postProcessBeforeInitialization() method")
+    void trimmedAnnotationBeanPostProcessor_notOverridesPostProcessAfterInitMethod() {
+        boolean isBeanPostProcessorPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getInterfaces())
+                .anyMatch(i -> i.isAssignableFrom(
+                        BeanPostProcessor.class));
+        if (!isBeanPostProcessorPresent) {
+            throw new AssertionError("TrimmedAnnotationBeanPostProcessor must implement BeanPostProcessor interface");
+        }
+
+        boolean isBeforeProcessingMethodPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getDeclaredMethods())
+                .anyMatch(m -> m.getName().equals("postProcessBeforeInitialization"));
+
+        assertFalse(isBeforeProcessingMethodPresent);
+    }
+
+    @Test
+    @Order(12)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor trims String input params marked by @Trimmed if " +
+            "trimming is enabled by @EnableStringTrimming ")
+    void trimmedAnnotationPostProcessor_annotatedInputParams(@Autowired TrimmedService service) {
+        String inputArgs = " Simba Bimba  ";
+        String actual = service.getTheTrimmedString(inputArgs);
+
+        assertEquals(inputArgs.trim(), actual);
+    }
+
+    @Test
+    @Order(13)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor not trims String input params that not marked by @Trimmed")
+    void trimmedAnnotationPostProcessor_notAnnotatedInputParams(@Autowired TrimmedService service) {
+        checkIfBeanPostProcessorImplemented();
```

Comment:
@Stanislav-Zabramnyi it looks like again this test tests more than one thing. Let's verify that BPP is implemented and the method is overridden in other tests. Please remove `checkIfBeanPostProcessorImplemented();` from here